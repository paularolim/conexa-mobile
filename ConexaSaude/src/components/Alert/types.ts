export type AlertType = 'error' | 'warning';

export interface AlertProps {
  type: AlertType;
  message: string;
}
