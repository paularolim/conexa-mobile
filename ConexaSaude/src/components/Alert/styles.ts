import { View } from 'react-native';
import styled from 'styled-components/native';

import { Text } from '@components/Text';

import { AlertType } from './types';

export const Container = styled(View)<{ type: AlertType }>`
  background-color: ${({ theme, type }) =>
    type === 'error' ? theme.colors.redLight : theme.colors.yellowLight};
  padding: 8px;
  border-radius: 4px;
`;

export const MessageLabel = styled(Text)<{ type: AlertType }>`
  font-size: 18px;
  color: ${({ theme, type }) => (type === 'error' ? theme.colors.red : theme.colors.yellow)};
`;
