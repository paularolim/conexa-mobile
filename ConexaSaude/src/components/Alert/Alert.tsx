import { Container, MessageLabel } from './styles';
import { AlertProps } from './types';

export function Alert({ type, message }: AlertProps) {
  return (
    <Container type={type}>
      <MessageLabel type={type}>{message}</MessageLabel>
    </Container>
  );
}
