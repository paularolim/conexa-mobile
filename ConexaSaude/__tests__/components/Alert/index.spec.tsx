import renderer from 'react-test-renderer';

import { Alert } from '@components/Alert';

import { Wrapper } from '../../Wrapper';

describe('Alert component', () => {
  it('should render component with type="error" correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <Alert type="error" message="Error message" />
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render component with type="warning" correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <Alert type="warning" message="Warning message" />
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
