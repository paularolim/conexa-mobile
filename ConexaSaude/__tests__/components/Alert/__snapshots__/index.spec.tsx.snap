// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component snapshots should render component with type="error" correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFEBEE",
        "borderRadius": 4,
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
      },
    ]
  }
  type="error"
>
  <Text
    style={
      [
        {
          "color": "#000000",
        },
        {
          "color": "#ED4337",
          "fontSize": 18,
        },
      ]
    }
    type="error"
  >
    Error message
  </Text>
</View>
`;

exports[`Alert component snapshots should render component with type="warning" correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFF4EC",
        "borderRadius": 4,
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
      },
    ]
  }
  type="warning"
>
  <Text
    style={
      [
        {
          "color": "#000000",
        },
        {
          "color": "#B95000",
          "fontSize": 18,
        },
      ]
    }
    type="warning"
  >
    Warning message
  </Text>
</View>
`;
